pkg/fake_gcloud/lib/mem_storage.dart

```diff
@@ -0,0 +1,223 @@
+  })  : crc32CChecksum = content.fold<int>(0, (a, b) => a + b) & 0xffffffff,
+        updated = DateTime.now().toUtc();
+
+  @override
+  Uri get downloadLink => Uri(scheme: 'gs', host: bucketName, path: name);
+
+  @override
+  String get etag => crc32CChecksum.toRadixString(16);
+
+  @override
+  ObjectGeneration get generation {
+    throw UnimplementedError();
+  }
+
+  @override
+  int get length => content.length;
+
+  @override
+  List<int> get md5Hash => etag.codeUnits;
+
+  @override
+  ObjectMetadata get metadata {
+    throw UnimplementedError();
+  }
+}
+
+class _Bucket implements Bucket {
+  final _files = <String, _File>{};
+
+  @override
+  final String bucketName;
+
+  _Bucket(this.bucketName);
+
+  @override
+  String absoluteObjectName(String objectName) {
+    _validateObjectName(objectName);
+    return Uri(scheme: 'gs', host: bucketName, path: objectName).toString();
+  }
+
+  @override
+  StreamSink<List<int>> write(String objectName,
+      {int length,
+      ObjectMetadata metadata,
+      Acl acl,
+      PredefinedAcl predefinedAcl,
+      String contentType}) {
+    _validateObjectName(objectName);
+    // ignore: close_sinks
+    final controller = StreamController<List<int>>();
+    controller.stream.fold<List<int>>(<int>[], (buffer, data) {
+      buffer.addAll(data);
+      return buffer;
+    }).then((content) {
+      _files[objectName] = _File(
+        bucketName: bucketName,
+        name: objectName,
+        content: content,
+      );
+    });
+    return controller.sink;
+  }
+
+  @override
+  Future<ObjectInfo> writeBytes(String name, List<int> bytes,
+      {ObjectMetadata metadata,
+      Acl acl,
+      PredefinedAcl predefinedAcl,
+      String contentType}) async {
+    _validateObjectName(name);
+    final sink = write(
+      name,
+      metadata: metadata,
+      acl: acl,
+      predefinedAcl: predefinedAcl,
+      contentType: contentType,
+    );
+    sink.add(bytes);
+    await sink.close();
+    return _files[name];
+  }
+
+  @override
+  Future delete(String name) async {
+    _validateObjectName(name);
+    _files.remove(name);
+  }
+
+  @override
+  Future<ObjectInfo> info(String name) async {
+    _validateObjectName(name);
+    final info = _files[name];
+    if (info == null) {
+      throw DetailedApiRequestError(404, '$name does not exists');
+    }
+    return info;
+  }
+
+  @override
+  Stream<List<int>> read(String objectName, {int offset, int length}) async* {
+    _validateObjectName(objectName);
+    final file = _files[objectName];
+    yield file.content;
+  }
+
+  @override
+  Stream<BucketEntry> list({String prefix}) async* {
+    _validateObjectName(prefix, allowNull: true);
+    for (String name in _files.keys) {
+      if (prefix == null || name.startsWith(prefix)) {
+        yield _BucketEntry(name, true);
+      }
+    }
+  }
+
+  @override
+  Future<Page<BucketEntry>> page({String prefix, int pageSize = 50}) async {
+    throw UnimplementedError();
```

Comment:
Same for all these.. Just something like: `throw UnimplementedError('fake_gcloud.Bucket.page is not implemented ');`